lib/developer_portal/app/controllers/developer_portal/admin/account/passwords_controller.rb

```diff
@@ -46,13 +46,19 @@ def redirect_to_request_password(error_message)
   end
 
   def buyer
-    @buyer ||= @provider.buyers.build
+    @buyer ||= @provider.buyers.build.tap do |account|
+      account.unflattened_attributes = spam_protection_params
+    end
   end
 
   def password_params
     params.require(:user).permit(:password, :password_confirmation)
   end
 
+  def spam_protection_params
```

Comment:
To me the name is a bit confusing, as it's fetching all the `:account` params, whether they are spam protection-related or not. At this point all of them are spam protection-related indeed.

I think I'd prefer to just call it `account_params`, and add a comment that fetching these params is for the spam protection mechanism. But well, maybe it's just me :upside_down_face:

Reply:
OK!